src/rdiff_backup/log.py

```diff
@@ -26,6 +26,17 @@
 import os  # needed to grab verbosity as environment variable
 from . import Globals, rpath
 
+LOGFILE_ENCODING = 'utf-8'
+
+
+def _to_bytes(logline, encoding=LOGFILE_ENCODING):
+    """
+    Convert string into bytes for logging into file.
+    """
+    assert logline
```

Comment:
It would be valid to have an empty string, which would fail here. You could check for None, but that would be IMHO lost CPU cycles.

Reply:
This is not a plausible case. The log line cannot be empty or none at this point. it's always call at least with "\n"